combat/CombatDamage.cpp

```diff
@@ -0,0 +1,211 @@
+        std::vector<float> retval;
+        if (!ship || !design)
+            return retval;
+        const std::vector<std::string>& parts = design->Parts();
+        if (parts.empty())
+            return retval;
+
+        MeterType METER = max ? MeterType::METER_MAX_CAPACITY : MeterType::METER_CAPACITY;
+        MeterType SECONDARY_METER = max ? MeterType::METER_MAX_SECONDARY_STAT : MeterType::METER_SECONDARY_STAT;
+
+        float fighter_damage = 0.0f;
+        int fighter_launch_capacity = 0;
+        int available_fighters = 0;
+
+        retval.reserve(parts.size() + 1);
+        int num_bouts = GetGameRules().Get<int>("RULE_NUM_COMBAT_ROUNDS");
+        // for each weapon part, get its damage meter value
+        for (const auto& part_name : parts) {
+            const ShipPart* part = GetShipPart(part_name);
+            if (!part)
+                continue;
+            ShipPartClass part_class = part->Class();
+
+            // get the attack power for each weapon part.
+            if (part_class == ShipPartClass::PC_DIRECT_WEAPON) {
+                if (target_ships)
+                    retval.emplace_back(ship->WeaponPartShipDamage(part, context));
+                else
+                    retval.emplace_back(ship->WeaponPartFighterDamage(part, context));
+            } else if (part_class == ShipPartClass::PC_FIGHTER_BAY && include_fighters) {
+                // launch capacity determined by capacity of bay
+                fighter_launch_capacity += static_cast<int>(ship->CurrentPartMeterValue(METER, part_name));
+
+            } else if (part_class == ShipPartClass::PC_FIGHTER_HANGAR && include_fighters) {
+                // attack strength of a ship's fighters per bout determined by the hangar...
+                // assuming all hangars on a ship are the same part type...
+                if (target_ships && part->TotalShipDamage()) {
+                    retval.emplace_back(part->TotalShipDamage()->Eval(context));
+                    // as TotalShipDamage contains the damage from all fighters, do not further include fighter
+                    include_fighters = false;
+                } else if (part->TotalFighterDamage() && !target_ships) {
+                    retval.emplace_back(part->TotalFighterDamage()->Eval(context));
+                    // as TotalFighterDamage contains the damage from all fighters, do not further include fighter
+                    include_fighters = false;
+                } else if (part->CombatTargets() && context.effect_target && part->CombatTargets()->Eval(context, context.effect_target)) {
+                    fighter_damage = ship->CurrentPartMeterValue(SECONDARY_METER, part_name);
+                    available_fighters = std::max(0, static_cast<int>(ship->CurrentPartMeterValue(METER, part_name)));  // stacked meter
+                } else {
+                    std::vector<const Condition::Condition*> target_conditions;
```

Comment:
could directly initialized `target_conditions` with a single pointer, rather than defalut init and then push

Reply:
target_conditions was a left-over; removed